telegram/src/main/kotlin/com/github/kotlintelegrambot/dispatcher/handlers/InputChainHandler.kt

```diff
@@ -0,0 +1,65 @@
+package com.github.kotlintelegrambot.dispatcher.handlers
+
+import com.github.kotlintelegrambot.Bot
+import com.github.kotlintelegrambot.entities.Message
+import com.github.kotlintelegrambot.entities.Update
+import com.github.kotlintelegrambot.extensions.filters.Filter
+
+data class InputChainHandlerEnvironment<T>(
+    val inputChainIdGen: (previousInputChainId: T?, Update) -> T?,
+    val inputChainId: T,
+    val bot: Bot,
+    val update: Update,
+    val message: Message,
+)
+
+class InputChainHandler<T>(
+    private val filter: Filter,
+    private val initialHandler: Handler,
+    private val inputChainIdGen: (previousInputChainId: T?, Update) -> T?,
+    private val handleInputChain: HandleInputChain<T>,
+) : Handler {
+    private val chainMap = mutableMapOf<Long, T?>()
```

Comment:
Also it looks like the value of the map is not really relevant, should this be a hash set of the visited "userId,chatId" instead? That way we can also get rid of `inputChainIdGen`. Or is there another reason why we'd like to have that in the environment object?

Reply:
The `T` value is the identifier of the "Input Question".
When the "chain action" is executed, it needs to know "which id it comes from".
Imagine you are creating a form with 3 questions like the one I posted in the PR description, but whenever the question is responded we want to store them on a DB. We need to know which question the user responded, and the value of the `T` is the id used. I used a generic type to allow devs to include their own data type (Maybe more than an int/String value is needed to continue the chain).